tensorflow/compiler/tf2tensorrt/convert/ops/fill_ops.cc

```diff
@@ -0,0 +1,165 @@
+    }
+    if (nbDims < 0) {
+        return errors::InvalidArgument("The dims parameter shape of the ", params.node_def.op(),
+                                        " operation in ", params.node_def.name(),
+                                        " cannot be partial.");
+    }
+    return Status::OK();
+  }
+
+  Status Convert() {
+    const auto &params = *this->params_;
+    const auto& inputs = params.inputs;
+    auto *converter = params.converter;
+    auto *network = converter->network();
+
+    const bool is_dims_static = inputs[0].is_weights();
+    const bool is_value_static = inputs[1].is_weights();
+
+    const TRT_TensorOrWeights& dims_input = inputs.at(0);
+    const TRT_TensorOrWeights& value_input = inputs.at(1);
+
+    int nbDims;
+    nvinfer1::Dims trt_dims;
+    if (is_dims_static) {
+      const auto dims_weights = dims_input.weights();
+      nbDims = dims_weights.Shape().dim(0);
+      trt_dims.nbDims = nbDims;
+      nvinfer1::DataType dims_type;
+      dims_type = dims_weights.TrtDType();
+      // Validate enforces dims_type == nvinfer1::DataType::kINT32
+      const int32* weight_ptr = dims_weights.GetPointer<int32>();
+      for (int i = 0; i < nbDims; ++i) {
+          trt_dims.d[i] = weight_ptr[i];
+      }
+    } else {
+      // trt_dims is empty, the layer will use input 0
+      trt_dims.nbDims = 0;
+      const auto dims_tensor = dims_input.tensor();
+      nbDims = dims_tensor->getDimensions().d[0];
+    }
+
+    // TensorRT IFillLayer requires a rank 0 scalar
+    ITensorProxyPtr scalar_tensor;
+    nvinfer1::Dims scalar_dims;
+    scalar_dims.nbDims = 0;
+    nvinfer1::DataType value_type;
+    if (is_value_static) {
+      value_type = value_input.weights().TrtDType();
+      scalar_tensor = converter->CreateConstantLayer(
+          value_input.weights(), scalar_dims);
+    } else {
+      value_type = value_input.tensor()->getType();
+      nvinfer1::IShuffleLayer* reshape =
+          network->addShuffle(*value_input.tensor()->trt_tensor());
+      TFTRT_RETURN_ERROR_IF_NULLPTR(reshape, params.node_def.name());
+      converter->SetLayerName(reshape, params.node_def, "reshape");
+      reshape->setReshapeDimensions(scalar_dims);
+      scalar_tensor = reshape->getOutput(0);
+    }
+
+    ITensorProxyPtr empty_beta_tensor;
+    nvinfer1::Dims beta_dims;
+    beta_dims.nbDims = 1;
+    beta_dims.d[0] = nbDims;
+    TF_RETURN_IF_ERROR(CreateScalarConstant(&params, 0, &empty_beta_tensor,
+                                            value_type,
+                                            beta_dims));
+
+    nvinfer1::IFillLayer* layer = network->addFill(
+        trt_dims,
+        nvinfer1::FillOperation::kLINSPACE);
+    TFTRT_RETURN_ERROR_IF_NULLPTR(layer, params.node_def.name());
+    if (!is_dims_static) {
+      layer->setInput(0, *dims_input.tensor()->trt_tensor());
+    }
+    layer->setInput(1, *scalar_tensor->trt_tensor());
+    layer->setInput(2, *empty_beta_tensor->trt_tensor());
```

Comment:
Isn't this only needed if value is a tensor (not a weight)? You could use setAlpha(), setBeta() otherwise.

Reply:
setAlpha and setBeta can be used only when the output is flat (not ND)